packages/spacecat-shared-utils/src/tracing-fetch.js

```diff
@@ -145,21 +190,22 @@ export async function tracingFetch(url, options) {
     setTraceHeaders(request, parentSegment, subSegment);
   }
 
-  const capturedAdobeFetch = async () => {
-    let response = null;
-    try {
-      response = await adobeFetch(request);
-    } catch (e) {
-      handleSubSegmentError(subSegment, request, e);
-      throw e;
-    }
+  subSegment.addAnnotation('timeout_ms', timeout);
 
-    setSubSegmentFlagsByStatusCode(subSegment, response.status);
+  try {
+    // Create a new request with the signal
+    const requestWithSignal = new Request(request, { signal });
+
+    // Use the same fetchWithTimeout function but catch errors to handle subsegment
+    const response = await fetchWithTimeout(requestWithSignal, null, signal);
```

Comment:
how about passing an empty object instead of `null`? Spread operation on null might fail

`fetchWithTimeout(requestWithSignal, {}, signal)`